bazel/cc_toolchains/clang_cc_toolchain_config.bzl

```diff
@@ -634,6 +642,10 @@ def _impl(ctx):
         ],
     )
 
+    whole_archive_flag = "-Wl,-whole-archive"
+    if ctx.attr.target_cpu in ["darwin", "darwin_arm64"]:
+        whole_archive_flag = "-Wl,-force_load"
```

Comment:
Unrelated change, or just something I missed in the description?

Reply:
See the macos build error, I'm trying to fix it. I'm guessing this wasn't really used.

Reply:
Ahh. I see.

I don't think this is quite right sadly... the `-force_load` ld64 flag works differently from the `-whole-archive` one in a big way. It may silently work, but I don't think it'll be doing what is intended.

I think I see how to fix this part of the toolchain config, lemme try it out.

Reply:
After reading through some old Java source code that handled this (the first reasonable example I could find) and some of the manual pages, I think something like this would be a fairly principled approach to make this work well with ld64:

https://github.com/chandlerc/carbon-lang/commit/d96a7164150cf62743a0ef19eb12527f322ff5fa

This at least builds for me on macOS (arm64 though).

Happy for you to cherry-pick that in here, or either of us to land it as a separate PR to clean up the flag handling. This

Reply:
Trying this out (pushed it separately while I look again at other parts of this)